### DISCUSSION TITLE
i need help reveiwing this code to see if it works properly

### DISCUSSION THREAD

**Discussion by authorofbelethrun** [Monika-After-Story/MonikaModDev#7920] [general - submod / external content]
so i am trying to write a random event where monika will ask the player if they have read fantasy novels, if the player answers a certain way monika will ask another question. I just want to make sure that the code is good before i do anything else to it 

```renpy
init 5 python:
    addEvent(
        Event(
            persistent.event_database,
            eventlabel="monika_harry_potter",
            category=["literature"],
            prompt="harry potter",
            random=True,
            

        )
    )


label monika_twitter:
  m 1rud "[player] , I have a question for you"
    m 3eud "do you like fantasy novels?"



m expression "Question?{nw}"
$ _history_list.pop()
menu:
    m "Question?{fast}"

    "yes":
        
m 3wub "Really that is wonderful [player] !"
m 2esb "Fantasy is something that I myself love"
m "especially Harry potter"
m 3ekb "do you know about Harry Potter [player] {nw}"


$ _history_list.pop()
menu:
    m "Question?{fast}"

  
    "No I don't":
        m 3ekd "aww that's a shame it is a really good series"
m 1esa "well I guess that means I'll just have to introduce you to the series"




"Only a muggle wouldn't know what Harry Potter is":
        #Dialogue/code for this path
    
    return








    "no":
        m 1ekt  "aww that's a shame, [player] !"
m 2esb "Fantasy is something that I myself love"
m "especially novels like Harry potter"
m "oh well i guess fantasy isn't for everyone "
return
```

**Comment by multimokia** [comment]
Okay, I see a bunch of things here. Let me condense this down firstly:
```renpy
init 5 python:
    addEvent(
        Event(
            persistent.event_database,
            eventlabel="monika_harry_potter",
            category=["literature"],
            prompt="harry potter",
            random=True,
        )
    )

label monika_twitter:
  m 1rud "[player] , I have a question for you"
    m 3eud "do you like fantasy novels?"

m expression "Question?{nw}"
$ _history_list.pop()
menu:
    m "Question?{fast}"

    "yes":
        
m 3wub "Really that is wonderful [player] !"
m 2esb "Fantasy is something that I myself love"
m "especially Harry potter"

m 3ekb "do you know about Harry Potter [player] {nw}"
$ _history_list.pop()
menu:
    m "Question?{fast}"

    "No I don't":
        m 3ekd "aww that's a shame it is a really good series"
m 1esa "well I guess that means I'll just have to introduce you to the series"

"Only a muggle wouldn't know what Harry Potter is":
        #Dialogue/code for this path
    
    return

    "no":
        m 1ekt  "aww that's a shame, [player] !"
m 2esb "Fantasy is something that I myself love"
m "especially novels like Harry potter"
m "oh well i guess fantasy isn't for everyone "
return
```

With the excess spaces gone, let's point out some issues:
```renpy
init 5 python:
    addEvent(
        Event(
            persistent.event_database,
            eventlabel="monika_harry_potter",
            category=["literature"],
            prompt="harry potter",
            random=True,
        )
    )

label monika_twitter:
```

label mismatch. You mention the label `monika_harry_potter`, this label is `monika_twitter`. It should be `label monika_harry_potter:` instead

Next:
```renpy
  m 1rud "[player] , I have a question for you"
    m 3eud "do you like fantasy novels?"
```

Indentation mismatch. These ***must*** be consistently spaced. We keep a standard of `4` spaces.

This then becomes
```renpy
    m 1rud "[player], I have a question for you."
    m 3eud "Do you like fantasy novels?"
```

Next:
```renpy
m expression "Question?{nw}"
$ _history_list.pop()
menu:
    m "Question?{fast}"

    "yes":
        
m 3wub "Really that is wonderful [player] !"
m 2esb "Fantasy is something that I myself love"
m "especially Harry potter"
m 3ekb "do you know about Harry Potter [player] {nw}"
```

This entire menu is not inside the label block anymore as it's not indented whatsoever. The full thing needs to go forward. Additionally, the dialogue you have for liking fantasy should be inside the `yes` menu option. As such, it should look more like:

```renpy
    m expression "Question?{nw}" #Don't forget to change 'expression' here
    $ _history_list.pop()
    menu:
        m "Question?{fast}"

        "Yes.":
            m 3wub "Really that is wonderful [player] !"
            m 2esb "Fantasy is something that I myself love"
            m "especially Harry potter"

            m 3ekb "do you know about Harry Potter [player] {nw}"
            $ _history_list.pop()
            menu:
                m "Question?{fast}"

                #Where's the yes option?

                "No I don't":
                    m 3ekd "aww that's a shame it is a really good series"
                    m 1esa "well I guess that means I'll just have to introduce you to the series"

                "Only a muggle wouldn't know what Harry Potter is":
                #Dialogue/code for this path

        "No.":
            m 1ekt  "aww that's a shame, [player] !"
            m 2esb "Fantasy is something that I myself love"
            m "especially novels like Harry potter"
            m "oh well i guess fantasy isn't for everyone "
            #You don't need a return here. We fall through to the bottom anyway
    return
```
I've also noticed you have some repeated dialogue:
```renpy
            m 2esb "Fantasy is something that I myself love"
            m "especially novels like Harry potter"
```

Normally we'd put this in its own label so it's better to maintain.
```renpy
label monika_harry_potter_love_fantasy:
    m 2esb "Fantasy is something that I myself love"
    m "especially novels like Harry potter"
    return
```

which cleans dialogue to be:
```renpy
    m expression "Question?{nw}" #Don't forget to change 'expression' here
    $ _history_list.pop()
    menu:
        m "Question?{fast}"

        "Yes.":
            m 3wub "Really that is wonderful [player] !"
            call monika_harry_potter_love_fantasy

            m 3ekb "do you know about Harry Potter [player] {nw}"
            $ _history_list.pop()
            menu:
                m "Question?{fast}"

                #Where's the yes option?

                "No I don't":
                    m 3ekd "aww that's a shame it is a really good series"
                    m 1esa "well I guess that means I'll just have to introduce you to the series"

                "Only a muggle wouldn't know what Harry Potter is":
                #Dialogue/code for this path

        "No.":
            m 1ekt  "aww that's a shame, [player] !"
            call monika_harry_potter_love_fantasy
            m "oh well i guess fantasy isn't for everyone "
            #You don't need a return here. We fall through to the bottom anyway
    return
```

There's also grammatical issues and other consistency issues I see, but I'll chalk that up to it being WIP right now.

**Comment by authorofbelethrun** [comment]
thank you so much!!

**Comment by multimokia** [comment]
Don't be afraid to test for yourself as well.

The game will report errors in your code when you launch it with your code in it.

Just be sure to set up an install for testing. [Check here to see how](https://github.com/Monika-After-Story/MonikaModDev/wiki/Contributing-Guidelines#bug-fixes-and-new-software-features)

**Comment by authorofbelethrun** [comment]
soo what would the full thing look like, because i plan to add a few more things to it.

**Comment by authorofbelethrun** [comment]
init 5 python:
    addEvent(
        Event(
            persistent.event_database,
            eventlabel="monika_harry_potter",
            category=["literature"],
            prompt="harry potter",
            random=True,
            

        )
    )
	
	
	label monika_harry_potter_love_fantasy:
    m 2esb "Fantasy is something that I myself love"
    m "especially novels like Harry potter"
    return


label monika_harry_potter:
    m 1rud "[player] , I have a question for you"
    m 3eud "do you like fantasy novels? {nw}"
    $ _history_list.pop()
    menu:
        m "Question?{fast}"

        "Yes.":
            m 3wub "Really that is wonderful [player] !"
            call monika_harry_potter_love_fantasy

            m 3ekb "do you know about Harry Potter [player]? {nw}"
            $ _history_list.pop()
            menu:
                m "Question?{fast}"

                

                "No I don't":
                    m 3ekd "aww that's a shame it is a really good series"
                    m 1esa "well I guess that means I'll just have to introduce you to the series"

                "Only a muggle wouldn't know what Harry Potter is":
                m 3wub "well that is amazing [player] , but now i have only one question for you"
				m      " what is your hogwarts house "

        "No.":
            m 1ekt  "aww that's a shame, [player] ..."
            call monika_harry_potter_love_fantasy
            m "oh well i guess fantasy isn't for everyone "
            #You don't need a return here. We fall through to the bottom anyway
    return



is this good???

**Comment by authorofbelethrun** [comment]
init 5 python:
    addEvent(
        Event(
            persistent.event_database,
            eventlabel="monika_harry_potter",
            category=["literature"],
            prompt="harry potter",
            random=True,
            

        )
    )
	
	
	label monika_harry_potter_love_fantasy:
    m 2esb "Fantasy is something that I myself love"
    m "especially novels like Harry potter"
    return


label monika_harry_potter:
    m 1rud "[player] , I have a question for you"
    m 3eud "do you like fantasy novels? {nw}"
    $ _history_list.pop()
    menu:
        m "Question?{fast}"

        "Yes.":
            m 3wub "Really that is wonderful [player] !"
            call monika_harry_potter_love_fantasy

            m 3ekb "do you know about Harry Potter [player]? {nw}"
            $ _history_list.pop()
            menu:
                m "Question?{fast}"

                

                "No I don't":
                    m 3ekd "aww that's a shame it is a really good series"
                    m 1esa "well I guess that means I'll just have to introduce you to the series"

                "Only a muggle wouldn't know what Harry Potter is":
                m 3wub "well that is amazing [player] , but now i have only one question for you"
				m      " what is your hogwarts house "
				return

        "No.":
            m 1ekt  "aww that's a shame, [player] ..."
            call monika_harry_potter_love_fantasy
            m "oh well i guess fantasy isn't for everyone "


here's a better version of it but im just curious if this works

**Comment by ThePotatoGuy** [comment]
if you are planning to submit this, its better to just make a PR. otherwise it's a bit of effort on our part to syntax check your code for something that won't be added to the mod. We're not personal tutors.
